### PR TITLE
Missing Function Download-BcEnvironmentInstalledExtensionToFolder (#3703)

### PR DESCRIPTION
**Pull request by zabcik**
Add Missing Function Download-BcEnvironmentInstalledExtensionToFolder

@microsoft-github-policy-service agree

### CODE REVIEW

**Comment by zabcik** [comment]
@microsoft-github-policy-service agree

File: Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1 [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        [Parameter(Mandatory = $true)]
+        [string] $appPublisher,
+        [Parameter(Mandatory = $true)]
+        [string] $appName,
+        [Parameter(Mandatory = $true)]        
+        [string] $appVersion = '0.0.0.0',
+        [Parameter(Mandatory = $false)]
+        [string] $appId
+    )
+    $telemetryScope = InitTelemetryScope -name $MyInvocation.InvocationName -parameterValues $PSBoundParameters -includeParameters @()
```
Comment (freddydk) [inline Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1:50]: Normally - I renew the authcontext inside the function.
Reason is that this function might be called deep inside a pipeline, which has been running for more than the validity of the accesstoken.
Then... - remove the renew in the sample above.

File: Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1 [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    try {
+        $bcAuthContext = Renew-BcAuthContext -bcAuthContext $bcAuthContext
+        $bearerAuthValue = "Bearer $($bcAuthContext.AccessToken)"
+        $headers = @{ "Authorization" = $bearerAuthValue }
+
+        #WARNING: The file name downloaded may not match the one on the server, as the server only returns the file's content. When requesting the file, I specify the name, publisher, version, and even the ID, but only the ID remains consistent.
+        $symbolsName = "$($appPublisher)_$($appName)_$($appVersion).app".Split([System.IO.Path]::GetInvalidFileNameChars()) -join ''
+        Write-Host "Downloading symbols: $symbolsName"
+
+        $symbolsFile = (Join-Path $appSymbolsFolder $symbolsName)
```
Comment (freddydk) [inline Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1:59]: The parameter is called $folder - not $appSymbolsFolder

File: Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1 [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        Write-Host "symbolsFile: $symbolsFile"
+
+        $devServerUrl = "$($bcContainerHelperConfig.apiBaseUrl.TrimEnd('/'))/$apiVersion/$environment"
+
+        if ($appId) {
+            $url = "$devServerUrl/dev/packages?appId=$($appId)&versionText=$($appVersion)&tenant=$($tenant)"
```
Comment (freddydk) [inline Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1:65]: $tenant is undefined (and not needed in the URL when connecting to an online environment)

**Review by freddydk — CHANGES_REQUESTED** [review]
Please also add a line to the release notes.

File: Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1 (hunk 3 above)
Comment (freddydk) [reply Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1:65]: BTW - tenant is used twice

**Review by freddydk — APPROVED** [review]

**Review by freddydk — APPROVED** [review]

**Review by aholstrup1 — APPROVED** [review]

File: Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1 [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        [string] $apiVersion = "v2.0",
+        [Parameter(Mandatory = $true)]
+        [alias('appSymbolsFolder')]
+        [string] $folder,
+        [Parameter(Mandatory = $true)]
+        [string] $appPublisher,
+        [Parameter(Mandatory = $true)]
+        [string] $appName,
+        [Parameter(Mandatory = $true)]        
+        [string] $appVersion = '0.0.0.0',
```
Comment (jonasblunck) [inline Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1:45]: Parameter is mandatory so doesn't make sense to add default value?

Comment (freddydk) [reply Saas/Download-BcEnvironmentInstalledExtensionToFolder.ps1:45]: Correct - will remove that in the next PR in order to get this one closed

**Pull request merged**